### PR TITLE
Resolve memebership and feature refs 

### PR DESCRIPTION
**Pull request by ziyue-101**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/controller/direct/gkehub/featuremembership_controller.go [hunk 1]
```diff
@@ -91,11 +91,11 @@ func (m *gkeHubModel) AdapterForObject(ctx context.Context, reader client.Reader
 	if projectID == "" {
 		return nil, fmt.Errorf("cannot resolve project")
 	}
-	mID, err := fullyQualifiedNameForMembership(*obj, projectID)
+	membership, err := resloveMembershipRef(ctx, reader, obj, projectID)
```
Comment (justinsb) [inline pkg/controller/direct/gkehub/featuremembership_controller.go:94]: Nit: typo in "resolve"

File: pkg/webhook/immutable_fields_validator.go [hunk 2]
```diff
@@ -328,27 +330,42 @@ func validateImmutableFieldsForTFBasedResource(obj, oldObj *unstructured.Unstruc
 	return allowedResponse
 }
 
+func isImmutableFieldsModified(oldSpec, newSpec map[string]interface{}, field string) bool {
```
Comment (justinsb) [inline pkg/webhook/immutable_fields_validator.go:333]: Nit: isImmutableFieldModified (it's only one field)

**Review by justinsb — APPROVED** [review]
Some nits, but lgtm!

**Comment by justinsb** [comment]
Two small naming nits, but not blockers.

/approve
/lgtm


/hold if you want to fix naming nits in this PR, feel free to cancel hold

**Comment by ziyue-101** [comment]
fixed the namings.  /unhold

**Comment by justinsb** [comment]
Thanks!

/approve
/lgtm

**Pull request merged**